xaitk_saliency/impls/gen_classifier_conf_sal/policy_scoring.py

```diff
@@ -0,0 +1,53 @@
+from xaitk_saliency import GenerateClassifierConfidenceSaliency
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+import numpy as np
+
+
+class PolicyScoring(GenerateClassifierConfidenceSaliency):
+    """
+    This saliency implementation transforms black-box action and value scores
+    from an actor-critic policy network into saliency heatmaps. This should
+    require a sequence of either action scores or values scores predicted on
+    the reference image, a number of action or values scores predicted on
+    perturbed images, as well as the masks of the reference image perturbations
+    (as would be output from a `PerturbImage` implementation).
+
+    Although the policy network is not an image classification model per se, it
+    shares certain similarities in that it also receives as input an image
+    (i.e. a video game frame) and outputs vectors of action and/or value scores
+    which are similar to classifier confidences.
+
+    Scoring is done using squared difference and the resulting maps are
+    normalized to be in the range [0,1].
+
+    Based on Greydanus et. al:
+    https://arxiv.org/abs/1711.00138
+    """
+
+    def generate(
+            self,
+            reference: np.ndarray,
+            perturbed: np.ndarray,
+            perturbed_masks: np.ndarray
+    ) -> np.ndarray:
+
+        if len(reference) != len(perturbed[0]):
+            raise ValueError("Number of classes in original image and"
+                             " perturbed image do not match.")
+
+        if len(perturbed) != len(perturbed_masks):
+            raise ValueError("Number of perturbation masks and respective "
+                             "confidence lengths do not match.")
+
+        # Based on equations 2 and 4 from Greydanus et al., '18
+        diff = 0.5*((reference - perturbed)**2).sum(axis=1, keepdims=True)
+        sal = weight_regions_by_scalar(diff, perturbed_masks)[0]
+
+        # Normalize saliency map to [0,1]
+        sal -= sal.min()
+        sal = sal / sal.max()
+
+        return sal
+
+    def get_config(self) -> dict:
```

Comment:
Looks like codecov complains about a missing test for the configuration portion of this new implementation- I think this should be relatively straightforward to add using a similar test from another implementation.